### PR TITLE
docs: add github icon on top of demo.tutorialkit.dev TopBar

### PR DESCRIPTION
**Pull request by Charan167**
- Add Github icon link on top of demo.tutorialkit.dev TopBar

https://github.com/user-attachments/assets/2b2aaa63-f491-4e0b-9e73-204e1e31d36f

### CODE REVIEW

**Review by AriPerkkio — CHANGES_REQUESTED** [review]
Great work @Charan167!

This PR should only make changes to `docs/demo/**` files. Couple of changes below:

- move the `Github.tsx` to `docs/demo/src/components/Github.tsx`
- add `docs/demo/src/components/TopBar.tsx` with following:
  ```astro
  ---
  import Github from "./Github";
  ---

  <nav
    class="bg-tk-elements-topBar-backgroundColor transition-theme border-b border-tk-elements-app-borderColor flex max-w-full items-center p-3 px-4 min-h-[56px]"
  >
    <div class="flex flex-1">
      <slot name="logo" />
    </div>
    <div class="mr-2">
      <slot name="open-in-stackblitz-link" />
    </div>

    <Github />

    <div>
      <slot name="theme-switch" />
    </div>
    <div>
      <slot name="login-button" />
    </div>
  </nav>
  ```
- define `components.TopBar` in `docs/demo/astro.config.ts` like this: https://github.com/AriPerkkio/tutorial-vite-plugin/pull/7/files#diff-baf06eda15601b59413c77e1f18884403033204061e6c9408cd9a3bb5bc6e586R10-R12

Now you should see the icon when running `pnpm dev` in `docs/demo` directory.

**Review by Charan167 — COMMENTED** [review]
- Moved Github.tsx to docs/demo/src/components/Github.tsx
- Defined components.TopBar in docs/demo/astro.config.ts

File: packages/astro/src/default/components/TopBar.astro [hunk 1]
```diff
@@ -7,6 +7,9 @@
   <div class="mr-2">
     <slot name="open-in-stackblitz-link" />
   </div>
+  <div class="mr-2">
+    <slot name="github" />
+  </div>
```
Comment (AriPerkkio) [inline packages/astro/src/default/components/TopBar.astro:12]:
```suggestion
```

**Review by AriPerkkio — APPROVED** [review]
Looks good, thanks @Charan167! 🎉 

If you want to work more on improving the demo site, good next task would be to add `editPageLink` there: https://tutorialkit.dev/reference/configuration/#editpagelink.

**Comment by Charan167** [comment]
@AriPerkkio Sure!!! I've made the changes by adding editpagelink in the lesson’s frontmatter block. Please confirm if it's supposed to work like this.

https://github.com/user-attachments/assets/87caf5f2-0c14-4c91-8562-015f6e5a93b8

**Comment by AriPerkkio** [comment]
Yes, that's the intention. But it looks like the link is not working. Maybe the URL is not correct?

Here's Tutorial that has correctly working edit page link: https://tutorial-vite-plugin.pages.dev

**Comment by Charan167** [comment]
@AriPerkkio Thank you for your help!!! I missed the demo part again 😅. I've made the necessary changes and created a PR.

**Pull request merged**